src/core/filter/security.go

```diff
@@ -194,6 +205,114 @@ func (r *robotAuthReqCtxModifier) Modify(ctx *beegoctx.Context) bool {
 	return true
 }
 
+type authProxyReqCtxModifier struct{}
+
+func (ap *authProxyReqCtxModifier) Modify(ctx *beegoctx.Context) bool {
+	// only support docker login
+	if ctx.Request.URL.Path == "/v2/" || ctx.Request.URL.Path == "/service/token" {
+		log.Error("You do not have the proper privileges to perform the action.")
+		return false
+	}
+
+	proxyUserName, proxyPwd, ok := ctx.Request.BasicAuth()
+	if !ok {
+		return false
+	}
+
+	match, rawUserName := ap.matchAuthProxyUserName(proxyUserName)
```

Comment:
conventionally the match should be second return value?  Refer to `value, ok = map[key]`